### PR TITLE
[IDE] Fix null reference when code uses Log.WriteLine

### PR DESCRIPTION
**Pull request by iainx**
The TextWriter was using the version of WriteConsoleLogText that takes no monitor. Change it to use the method that takes a monitor, but improve handling for
null monitors.

Fixes VSTS #567563

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs [hunk 1]
```diff
@@ -567,14 +567,17 @@ protected void UnsafeBeginTask (LogViewProgressMonitor monitor, string name, int
 				UnsafeAddText (null, Environment.NewLine, null);
 			}
 
-			monitor.Marker = buffer.CreateMark (name, buffer.EndIter, false);
+			var marker = buffer.CreateMark (name, buffer.EndIter, false);
+			if (monitor != null) {
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Do we need to add a similar null check in UnsafeEndTask?

https://github.com/mono/monodevelop/blob/5f04fee84c43f847b394261fbcdd4dcdfb012c87/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs#L606

Comment (mrward) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Maybe we don't. If monitor is null then we push a null so it should not go into that if statement.

Comment (iainx) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Doh, yes

Comment (iainx) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Done

**Pull request merged**